File: conjuntos-dicionarios/javascript/dict-test.js
let cookies = new Map();

cookies.set("user", "maria34");
cookies.set("email", "ana@gmail.com");
cookies.set("email", "maria@gmail.com"); // Sobrescreve o valor anterior
cookies.set("last-login", "2024-07-04T21:42:40.353283800Z");

console.log("Values:");
for (let value of cookies.values()) {
    console.log(value);
}

console.log("Keys & values:");
for (let [key, value] of cookies) {
    console.log(key + ": " + value);
}

console.log("Is empty: " + (cookies.size === 0));
console.log("Size: " + cookies.size);

console.log("Contains 'email': " + cookies.has("email"));
let removed = cookies.get("email");
console.log("Remove: " + removed);
cookies.delete("email");

console.log("Contains 'email': " + cookies.has("email"));
removed = cookies.get("email");
console.log("Remove: " + removed);

File: arvores/java/bst/src/DictTest.java
import bst.BinarySearchTreeMap;

public class DictTest {

	public static void main(String[] args) {
		BinarySearchTreeMap<String, String> cookies = new BinarySearchTreeMap<>();
		
		cookies.put("user", "maria34");
		cookies.put("email", "ana@gmail.com");
		cookies.put("email", "maria@gmail.com");
		cookies.put("last-login", "2024-07-04T21:42:40.353283800Z");
		
		System.out.println("Values:");
		for (String value : cookies.values()) {
			System.out.println(value);
		}
		
		System.out.println("\nKeys & values:");
		for (String key : cookies.keys()) {
			System.out.println(key + ": " + cookies.get(key));
		}
		
		System.out.println("\ntoString:");
		System.out.println(cookies);
		
		System.out.println("\nIs empty: " + cookies.isEmpty());
		System.out.println("Size: " + cookies.size());
		
		System.out.println("\nContains 'email': " + cookies.containsKey("email"));
		String obj = cookies.remove("email");
		System.out.println("Remove: " + obj);

		System.out.println("\nContains 'email': " + cookies.containsKey("email"));
		obj = cookies.remove("email");
		System.out.println("Remove: " + obj);
	}
}
